src/main/java/frc/robot/drivetrains/tankdrivetrain/Drivetrain.java

```diff
@@ -90,39 +102,56 @@ private Drivetrain() {
+                        new WPI_TalonSRX(RobotMap.CAN.DRIVETRAIN_LEFT_TALON_1),
+                        new WPI_TalonSRX(RobotMap.CAN.DRIVETRAIN_LEFT_TALON_2)
                 ),
                 new BustedMotorControllerGroup(
                         RIGHT_CORRECTIONS,
-                        new WPI_TalonSRX(RobotMap.CAN.RIGHT_TALON_1),
-                        new WPI_TalonSRX(RobotMap.CAN.RIGHT_TALON_2)
+                        new WPI_TalonSRX(RobotMap.CAN.DRIVETRAIN_RIGHT_TALON_1),
+                        new WPI_TalonSRX(RobotMap.CAN.DRIVETRAIN_RIGHT_TALON_2)
                 )
         );
-        this.leftEncoder = new Encoder(RobotMap.DIO.LEFT_ENCODER_A, RobotMap.DIO.LEFT_ENCODER_B);
-        this.rightEncoder = new Encoder(RobotMap.DIO.RIGHT_ENCODER_A, RobotMap.DIO.RIGHT_ENCODER_B);
+        this.leftEncoder = new Encoder(RobotMap.DIO.DRIVETRAIN_LEFT_ENCODER_A, RobotMap.DIO.DRIVETRAIN_LEFT_ENCODER_B);
+        this.rightEncoder = new Encoder(RobotMap.DIO.DRIVETRAIN_RIGHT_ENCODER_A, RobotMap.DIO.DRIVETRAIN_RIGHT_ENCODER_B);
         leftEncoder.setDistancePerPulse(DISTANCE_PER_PULSE);
         rightEncoder.setDistancePerPulse(DISTANCE_PER_PULSE);
         this.gyro = new ADXRS450_Gyro();
     }
 
-    public double getLeftDistance() {
-        return leftEncoder.getDistance();
+    public void resetEncoders() {
+        leftEncoder.reset();
+        rightEncoder.reset();
     }
 
-    public double getRightDistance() {
-        return rightEncoder.getDistance();
+    public void resetGyro() {
+        gyro.reset();
     }
 
     /**
-     * @return the robot's current angle in the range (0, infinity)
+     * Sets the values which appear on the shuffleboard using a {@link Namespace}.
+     */
+    public void configureDashboard() {
+        rootNamespace.putNumber("left distance", this::getLeftDistance);
+        rootNamespace.putNumber("right distance", this::getRightDistance);
+        rootNamespace.putNumber("gyro angle", this::getAngle);
+        rootNamespace.putNumber("gyro modified angle", this::getModifiedAngle);
+        rootNamespace.putData("reset encoders", new InstantCommand(this::resetEncoders));
+        rootNamespace.putData("reset gyro", new InstantCommand(this::resetGyro));
+        rootNamespace.putData("drive forward", new DriveArcade(this, DRIVE_SPEED, () -> 0.0));
+        rootNamespace.putData("drive backward", new DriveArcade(this, () -> -DRIVE_SPEED.get(), () -> 0.0));
+        rootNamespace.putData("drive meters", new DriveTankWithPID(this, pidSettings, pidSettings,
+                METERS_TO_DRIVE, METERS_TO_DRIVE, this::getLeftDistance, this::getRightDistance, ffSettings, ffSettings));
+    }
+
+    /**
+     * @return the robot's current angle in the range (-infinity, infinity)
      */
     public double getAngle() {
         return gyro.getAngle();
     }
 
     /**
-     * @return the robot's current angle in the range (-180, 180]
+     * @return the robot's current angle in the range [-180, 180]
```

Comment:
can `getModifiedAngle()` get a value of `-180`?

Reply:
yes

Reply:
how

Reply:
![image](https://user-images.githubusercontent.com/71877371/157022451-74187f82-faf6-4987-80ab-59658d4ebc93.png)

if the angle is `-180`, then `getModifiedAngle()` will return `-180`

Reply:
why the fuck does (-180)%360 not return 180? It should absolutely return 180. Java is wrong.

Reply:
That's not how a remainder works...

Reply:
It should return a value from 0 to 360, not including 360.

Reply:
anyway change it so it could only get 180, as 180 and -180 are the same thing